### ISSUE TITLE
GDB Segmentation Fault

### ISSUE THREAD

**Issue by bbouzan** [USCiLab/cereal#223]
Hi,

I'm facing an odd problem while debugging a program that uses Cereal with GDB.

I'm trying to serialize a class that has a vector of shared_ptr to a common Base class. But all objects are instances of classe the have up to three levels of inheritance from this Base class.

I can Serialize and Deserialize the class perfectly when running my program from the command line. But whenever I try to debug my program, the GDB crashes with segmentation fault every time 
````
void load(Archive& archive)
````
is not commented out.

The error I get from GDB is:
````
Reading symbols from /home/rvcpu/Source/RvcApplication/Debug/RvcApplication...Segmentation fault (core dumped)
````

Initialy I though it was related to issue #188, but I'm actaully not using Multiple Inheritance.

Anyone has an idea of what I'm doing wrong with GDB?

**Comment by AzothAmmo** [comment]
Chances are this is probably something on our end but I've been in similar situations (not necessarily with cereal) and usually can't figure out what makes GDB decide to crash.  If you can isolate a test case for this I can see if it crashes GDB on other machines and see if it doesn't get fixed inadvertently while working on some other issues.

**Comment by bbouzan** [comment]
Hi,

Thanks for the reply.

I tested the exact same code as the example on "Registering from a header file" section of 
http://uscilab.github.io/cereal/polymorphism.html

The GDB crashes when reading symbols. If I run the exact same executable from command line it runs fine.
Also, if I comment out the following portion:
```javascript
iarchive( ptr1, ptr2 );

// Ta-da! This should output:
ptr1->sayType();  // "DerivedClassOne"
ptr2->sayType();  // "EmbarrassingDerivedClass. Wait.. I mean DerivedClassTwo!"*/
```
the GDB problem doesn't happen.

My program is being cross-compiled to run on armv7, so I'm using:

GNU gdb (Sourcery CodeBench Lite 2014.05-29) 7.7.50.20140217-cvs
GNU gdbserver (Sourcery CodeBench Lite 2014.05-29) 7.7.50.20140217-cvs

**Comment by neiser** [comment]
I experienced a similar problem (gdb cannot properly demangle some more complicated C++11 names, see here: https://sourceware.org/bugzilla/show_bug.cgi?id=16957) and solved it by installing gdb 7.10. Maybe that helps.

**Comment by bbouzan** [comment]
Thanks @neiser.
Indeed using gdb 7.10 the problem was solved.
